### PR TITLE
Fix alignments for children of Query block.

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Children of the Query block weren't being passed all the data on their parent's layout, so they weren't getting the full set of alignment controls when Query had "Inner blocks use content width" set.

Fixes an [issue reported on TT3](https://github.com/WordPress/twentytwentythree/issues/164#issuecomment-1246126615).

## Testing Instructions

1. Add a Query block to a template;
2. Check that the child Post Template block has "full" and "wide" alignment options;
3. Check that setting Post Template to "wide" alignment works correctly in the editor and front end.

## Screenshots or screencast 
Screen Shot 2022-09-14 at 3 39 37 pm

### CODE REVIEW

**Review by andrewserong — APPROVED** [review]
Nice one, thanks for finding a fix for this @tellthemachines. Glad it was a simple change copying over the logic from the Group block! I wonder if longer term it's worth us creating a common layout utility function that wraps this logic for easy use (not a blocker for this PR, of course).

This is testing well for me; with this PR applied I can now see the additional alignment controls at the Post Template level and the classnames are output in the site editor:

image

LGTM! Also, once this PR is merged, should we cherry pick it into the `release/14.1` branch so it makes it into this GB release?

**Comment by tellthemachines** [comment]
Thanks for reviewing @andrewserong !

>  I wonder if longer term it's worth us creating a common layout utility function that wraps this logic for easy use

Ideally we can find a way to make this block support work without having to explicitly pass layout to the inner blocks, which would incidentally fix #43731 😄 If/when we do that, perhaps we can remove this logic altogether 🤞 

> should we cherry pick it into the release/14.1 branch so it makes it into this GB release?

Yes!

**Comment by CreativeDive** [comment]
@tellthemachines @andrewserong I'm constantly following the progress you're making here, still hoping that we'll get `wide` and `full` back for our inner blocks inside a custom container blocks. 🙏

**Comment by cbravobernal** [comment]
[Cherry-picked for 14.1](https://github.com/WordPress/gutenberg/commit/527ae1852857f499efd07e852d02384893fc4066)

**Comment by ockham** [comment]
I just cherry-picked this PR to the wp/6.1 branch to get it included in the next release: f70250a749

**Pull request merged**
